service/messageHandler_test.go

```diff
@@ -50,200 +42,113 @@ func (p *mockMessageProducer) Shutdown() {
+		},
+		"hasBrand": {
+			PredicateURI:    "http://www.ft.com/ontology/hasBrand",
+			PredicateMapped: "hasBrand",
 		},
-		Body: fmt.Sprintf(`{
-		"uuid":"%s",
-		"submittedBy":"test-user",
-		"annotations":[
-		    {
-		        "predicate":"%s",
-		        "id":"%s"
-		    },
-		    {
-		        "predicate":"%s",
-		        "id":"%s"
-		    },
-		    {
-		        "predicate":"%s",
-		        "id":"%s"
-		    },
-		    {
-		        "predicate":"%s",
-		        "id":"%s"
-		    },
-		    {
-		        "predicate":"%s",
-		        "id":"%s"
-		    },
-		    {
-		        "predicate":"%s",
-		        "id":"%s"
-		    },
-		    {
-		        "predicate":"%s",
-		        "id":"%s"
-		    }
-		]
-		}`, contentUuid,
-			hasBrand, brandUuid,
-			implicitlyClassifiedBy, implicitlyClassifiedByUuid,
-			hasAuthor, hasAuthorUuid,
-			hasContributor, hasContributorUuid,
-			about, aboutUuid,
-			hasDisplayTag, hasDisplayTagUuid,
-			mentions, mentionsUuid),
 	}
 
-	err := service.HandleMessage(inbound)
-	assert.NoError(t, err)
-
-	mp.AssertExpectations(t)
-
-	require.Len(t, mp.received, 1, "messages sent to producer")
-
-	actual := mp.received[0]
-	assert.Equal(t, testTxId, actual.Headers["X-Request-Id"], "transaction_id should be propagated")
-
-	actualBody := MappedAnnotations{}
-	json.NewDecoder(strings.NewReader(actual.Body)).Decode(&actualBody)
-
-	assert.Equal(t, contentUuid, actualBody.UUID, "content uuid")
-
-	actualAnnotations := actualBody.Annotations
-	assert.Len(t, actualAnnotations, 7, "annotations")
-
-	foundBrand := false
-	foundMentions := false
-	foundImplicitlyClassifiedBy := false
-	foundHasAuthor := false
-	foundHasContributor := false
-	foundAbout := false
-	foundHasDisplayTag := false
-
-	for _, ann := range actualAnnotations {
-		switch ann.Concept.Predicate {
-		case "isClassifiedBy":
-			assert.Equal(t, brandUuid, ann.Concept.ID, "brand annotation")
-			foundBrand = true
-
-		case "mentions":
-			assert.Equal(t, mentionsUuid, ann.Concept.ID, "mentions annotation")
-			foundMentions = true
-
-		case "implicitlyClassifiedBy":
-			assert.Equal(t, implicitlyClassifiedByUuid, ann.Concept.ID, "implicitlyClassifiedBy annotation")
-			foundImplicitlyClassifiedBy = true
-
-		case "hasAuthor":
-			assert.Equal(t, hasAuthorUuid, ann.Concept.ID, "hasAuthor annotation")
-			foundHasAuthor = true
-
-		case "hasContributor":
-			assert.Equal(t, hasContributorUuid, ann.Concept.ID, "hasContributor annotation")
-			foundHasContributor = true
-
-		case "about":
-			assert.Equal(t, aboutUuid, ann.Concept.ID, "about annotation")
-			foundAbout = true
-
-		case "hasDisplayTag":
-			assert.Equal(t, hasDisplayTagUuid, ann.Concept.ID, "hasDisplayTag annotation")
-			foundHasDisplayTag = true
-		}
-
+	for testName, test := range tests {
+		t.Run(testName, func(t *testing.T) {
+			mp := &mockMessageProducer{}
+			mp.On("SendMessage", mock.AnythingOfType("kafka.FTMessage")).Return(nil)
+
+			service := NewAnnotationMapperService(whitelist, mp)
```

Comment:
Why do we need to reinitialise the service on every test run?

Reply:
To minimize side effects from other tests. `mockMessageProducer{}` is a stateful and we were testing that state with `require.Len(t, mp.received, 1, "messages sent to producer")` .
I don't see muck value in that require test, but decided to keep it for consistency.